types/runtime.ts

```diff
@@ -83,5 +83,18 @@ export type GcpConfig = (
 ) & {
   cloud: "gcp";
   name?: string;
-  environmentOverrides?: Partial<Record<string, Partial<Omit<GcpConfig, "environmentOverrides">>>>;
+  environmentOverrides?: {
```

Comment:
The way you've re-defined the top level `environmentOverrides` object loosens the type more than originally intended - the `Partial<>` utility wrapping it was intentional.

When you define a record type with a catch-all key (either through `Record<keytype, valuetype>` or as an object `{ [key: keytype]: valuetype; }`), Typescript does not attempt to narrow it and will accept any key value as valid. This means that you could refer to non-existent keys in the record and won't get any complaints from TS about the type of the value it would return (note the retrieved value type in the screenshot is `string`, but in reality it's `undefined` because the key does not exist):
![image](https://github.com/user-attachments/assets/79cd3fbc-9af7-4b34-af8f-f07d3d50917c)

However, if you wrap that record type in a `Partial<>` utility, Typescript now require you to validate the existence of keys before using the value type (in this screenshot, the value type is correctly identified as `string | undefined`, because TS does not know if the key exists):
![image](https://github.com/user-attachments/assets/700058d3-ce53-46d5-afe1-86a3f1f2df01)

Reply:
Thank you for this, confirmed on slack a new approach ✅